modules/client/src/main/java/org/apache/ignite/internal/client/ReliableChannel.java

```diff
@@ -583,8 +588,14 @@ private ClientChannel getDefaultChannel() {
     private boolean shouldRetry(int opCode, int iteration, IgniteClientConnectionException exception) {
         ClientOperationType opType = ClientUtils.opCodeToClientOperationType(opCode);
 
+        return shouldRetry(opType, iteration, exception);
+    }
+
+    /** Determines whether specified operation should be retried. */
+    private boolean shouldRetry(ClientOperationType opType, int iteration, IgniteClientConnectionException exception) {
         if (opType == null) {
-            return true; // System operation.
+            // System operation.
+            return iteration < RetryLimitPolicy.DFLT_RETRY_LIMIT;
```

Comment:
So does `opType == null` is still used for some operations?

Reply:
Let's leave it this way for now, I'm not sure exposing too much internal implementation details to user will do any good here.